src_features/generic_tx_parser/gtp_field_table.c

```diff
@@ -0,0 +1,99 @@
+#ifdef HAVE_GENERIC_TX_PARSER
+
+#include <string.h>
+#include "os_print.h"
+#include "gtp_field_table.h"
+#include "mem.h"
+
+// type (1) | key_length (1) | value_length (2) | key ($key_length) | value ($value_length)
+
+typedef struct {
+    const uint8_t *start;
+    size_t size;
+} s_field_table;
+
+static s_field_table g_table;
+
+void field_table_init(void) {
+    explicit_bzero(&g_table, sizeof(g_table));
+}
+
+const uint8_t *get_next_table_entry(const uint8_t *ptr, s_field_table_entry *entry) {
+    uint8_t key_len;
+    uint16_t value_len;
+
+    if (entry != NULL) {
+        memcpy(&entry->type, ptr, sizeof(entry->type));
+    }
+    ptr += sizeof(entry->type);
+    memcpy(&key_len, ptr, sizeof(key_len));
+    ptr += sizeof(key_len);
+    memcpy(&value_len, ptr, sizeof(value_len));
+    ptr += sizeof(value_len);
+    if (entry != NULL) {
+        entry->key = (char *) ptr;
+    }
+    ptr += key_len;
+    if (entry != NULL) {
+        entry->value = (char *) ptr;
+    }
+    ptr += value_len;
+    return ptr;
+}
+
+void field_table_cleanup(void) {
```

Comment:
Can you confirm that no other allocations are made between the calls to  `add_to_field_table` and the call to `field_table_cleanup`?

Reply:
Indeed, no other allocation. :+1: